the-collab-lab/tcl-27-smart-shopping-list
# Allow user to check off an item that is purchased

## Description

* Purchased checkbox is disabled after 24 hours
* Purchased timestamp is deleted from database after 24 hours
* Purchased is set to false in databased after 24 hours

## Related Issue

closes #8 

## Acceptance Criteria
[AC](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/8)

AC:
* User is able to tap a checkbox or similar UI element to mark an item in the list as purchased
* Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From your terminal, pull from main branch with git pull origin main
* Pull this branch with git pull origin sk-dp-check-as-purchased
* Checkout branch with git checkout sk-dp-check-as-purchased

Add an item to the list and then check the checkbox to state that the item is purchased. Go to [firestore](https://console.firebase.google.com/project/tcl-27-shopping-list/firestore/data/~2Fgroceries) and make sure that the submitted item has a purchased boolean value and a checkedTime timestamp value. Edit the checkedTime value to a date/time that is over 24 hours ago and verify that the checkbox in the list is unchecked after refreshing page. Check database one more time and verify that the lastPurchase date has been updated to be the previous checkedTime date/time.
